### ISSUE TITLE
EKS AccessControlListNotSupported: The bucket does not allow ACLs

### ISSUE THREAD

**Issue by tgelpi-gmail** [jenkins-x/jx#8572]
When creating a new EKS environment using OOTB config you encounter errors regarding long term storage creation
```
│ Error: error creating S3 bucket ACL for logs-jx3fri-20230421172407921400000009: AccessControlListNotSupported: The bucket does not allow ACLs
│ 	status code: 400, request id: FBT5A0MKD0FFVN7K, host id: 7FR0M1P79alIyR0mT5YxbKuaxtuLdVsoLkhwa5Avxpkmz/ZenqArzwx4lxZuKDTgtOrdz4fOXRk=
│ 
│   with module.eks-jx.module.cluster.aws_s3_bucket_acl.logs_jenkins_x[0],
│   on .terraform/modules/eks-jx/modules/cluster/storage.tf line 22, in resource "aws_s3_bucket_acl" "logs_jenkins_x":
│   22: resource "aws_s3_bucket_acl" "logs_jenkins_x" {
│ 
```
[A notice from Amazon indicates that Amazon S3 will automatically enable S3 Block Public Access and disable access control lists for all new buckets starting in April 2023](https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-s3-automatically-enable-block-public-access-disable-access-control-lists-buckets-april-2023/)

Currently storage allocation defaults to ACL
```
resource "aws_s3_bucket_acl" "logs_jenkins_x" {
  count  = var.enable_logs_storage ? 1 : 0
  bucket = aws_s3_bucket.logs_jenkins_x[0].bucket
  acl    = "private"
}
```

 Tried correcting this issue by updating  “**aws_s3_bucket_acl**” resource with “**aws_s3_bucket_ownership_controls**” resource setting the object_ownership rule to “**BucketOwnerEnforced**”

```
resource "aws_s3_bucket_ownership_controls" "logs_jenkins_x" {
  count  = var.enable_logs_storage ? 1 : 0
  bucket = aws_s3_bucket.logs_jenkins_x[0].bucket

  rule {
      object_ownership = "BucketOwnerEnforced"
    }
}
```
This type of storage update results in a clean build of the S3 buckets but it doesn't appear to be any content written to those buckets. Currently not clear on how to test and validate this change.

Using the latest version of cli, plugins, etc.

**Comment by tgelpi-gmail** [comment]
This issue impacts the creation of the following S3 buckets:
```
lts_logs_bucke
lts_reports_bucket 
lts_repository_bucket 
vault_unseal_bucket 
```

**Comment by tgelpi-gmail** [comment]
The issue potentially impacts the creation of 5 buckets
```
vault-unseal-bucket
logs_jenkins_x
reports_jenkins_x
repository_jenkins_x
backup_bucket
```
The files that need updating are:
```
modules/vault/main.tf
modules/cluster/storage.tf
modules/backup/main.tf
```
A new variable "**enable_acl**" is being considered for backward compatibility

**Comment by tgelpi** [comment]
I have a configuration that resolves this issue of ACL not supported by setting object ownership controls instead of ACL for the five S3 buckets.

**Comment by tgelpi** [comment]
This issue has been fixed with PR #362

**Comment by tgelpi-gmail** [comment]
This issue has been fixed with PR https://github.com/jenkins-x/terraform-aws-eks-jx/pull/362
